### DISCUSSION TITLE
What version of Python does pyttb require?

### DISCUSSION THREAD

**Discussion by tgkolda** [sandialabs/pyttb#336] [General]
Is there a minimum version for Python? How about associated libraries?

**Comment by tgkolda** [comment]
Also, what libraries does pyttb need?

**Comment by dmdunla** [comment]
> Is there a minimum version for Python? How about associated libraries?

As noted in the badges on the top of the README, pyttb supports supports 3.9, 3.10, 3.11, and 3.12:
https://github.com/sandialabs/pyttb

This Lython version info also appears on the PyPI front page on the left nav/info panel:
https://pypi.org/project/pyttb/

The dependencies and their supported versions appear in pyproject.toml file (used by pip for installs):
https://github.com/sandialabs/pyttb/blob/main/pyproject.toml

**Reply by tgkolda** [reply]
I'm not familiar with badges, though I was able to find it once you pointed me that way. I may have just been looking at the text version of the README.md file. Would it be reasonable to also state this there? For the packages, I was mainly curious, though it might be nice to give users an idea ahead of time of dependencies.

**Comment by dmdunla** [comment]
@tgkolda Can you describe your use case of where this info is needed and/or how you are using it? When installing with pip, the minimum Python version should be checked, and the dependency versions should also be checked so you should not need to track this down for manual alignment.

**Reply by tgkolda** [reply]
I had to pick my python version for my conda setup, which I was doing before pip install of pyttb.
